### PR TITLE
Add writes and fileMap for txn

### PR DESCRIPTION
**Pull request by nnsgmsone**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [X] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #4190

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/vm/engine/disttae/types.go [hunk 1]
```diff
@@ -17,19 +17,58 @@ package disttae
...
+	// fileMaps used to store the mapping relationship between s3 filenames and blockId
+	fileMap map[string]uint64
+	// writes cache stores any writes done by txn
+	// every statement is an element
+	writes   [][]Entry
+	dnStores []logservice.DNStore
+}
+
+// Entry represents a delete/insert
+type Entry struct {
```
Comment (fengttt) [inline pkg/vm/engine/disttae/types.go:68]: How can you tell if this entry is an insert of delete?

File: pkg/vm/engine/disttae/engine.go [hunk 2]
```diff
@@ -39,32 +40,40 @@ func New(
...
 func (e *Engine) Databases(ctx context.Context, op client.TxnOperator) ([]string, error) {
 	//TODO
-	e.newTransaction(op)
 	panic("unimplemented")
 }
 
 func (e *Engine) Delete(ctx context.Context, name string, op client.TxnOperator) error {
-	//TODO
-	e.newTransaction(op)
-	panic("unimplemented")
```
Comment (fengttt) [inline pkg/vm/engine/disttae/engine.go:62]: A transaction may have mixed insert/delete.     Do you want to separate them or combine them as "write"?

Comment (nnsgmsone) [reply pkg/vm/engine/disttae/engine.go:62]: My personal preference is to treat them all as write, but this is only a preference, so if people feel that it is better to separate them, it is just as well to do so. @fengttt

File: pkg/vm/engine/disttae/types.go (hunk 1 above)
Comment (nnsgmsone) [reply pkg/vm/engine/disttae/types.go:68]: My question, I missed the type. Embarrassment. I'll add it later

**Review by reusee — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

File: pkg/vm/engine/disttae/types.go (hunk 1 above)
Comment (fengttt) [reply pkg/vm/engine/disttae/types.go:73]: Where is rowid?   Inside batch?

Comment (nnsgmsone) [reply pkg/vm/engine/disttae/types.go:73]: yes, The data for rowid is a column.

File: pkg/vm/engine/disttae/types.go [hunk 3]
```diff
@@ -17,19 +17,64 @@ package disttae
...
+	typ          int
+	tableName    string
```
Comment (fengttt) [inline pkg/vm/engine/disttae/types.go:75]: table and database please use ID, not name.

File: pkg/vm/engine/disttae/types.go [hunk 4]
```diff
@@ -17,19 +17,64 @@ package disttae
...
+	databaseName string
+	fileName     string       // blockName for s3 file
```
Comment (fengttt) [inline pkg/vm/engine/disttae/types.go:77]: TAE people will let you know S3 name block id etc.  

Again, where is rowid?

Comment (nnsgmsone) [reply pkg/vm/engine/disttae/types.go:77]: in batch，exists as a column.

File: pkg/vm/engine/disttae/types.go (hunk 3 above)
Comment (nnsgmsone) [reply pkg/vm/engine/disttae/types.go:75]: why？Within a matter, names do not conflict。Of course it's the same with ids, just curious as to why.

**Pull request merged**